packages/SwingSet/src/kernel/vatManager/supervisor-helper.js

```diff
@@ -191,3 +191,45 @@ function makeSupervisorSyscall(syscallToManager, workerCanBlock) {
 
 harden(makeSupervisorSyscall);
 export { makeSupervisorSyscall };
+
+/**
+ * Create a vat console, or a no-op if consensusMode is true.
+ *
+ * TODO: consider other methods per SES VirtualConsole.
+ * See https://github.com/Agoric/agoric-sdk/issues/2146
+ *
+ * @param {Record<'debug' | 'log' | 'info' | 'warn' | 'error', (...args: any[]) => void>} logger
+ * the backing log method
+ * @param {boolean | ((logger: any, args: any[]) => void)} [wrapper]
+ */
+function makeVatConsole(logger, wrapper = true) {
+  const cons = Object.fromEntries(
+    ['debug', 'log', 'info', 'warn', 'error'].map(level => {
+      if (wrapper === false) {
+        // Static wrapper that never enables.  We create unique no-op functions
+        // so that the different log methods cannot be compared to detect this
+        // mode.
```

Comment:
I heartily approve of your paranoia :), but userspace code can always sense log-nop mode by `console.log({ toString: counter++})`, so it's probably not worth putting a lot of energy into trying to hide it.